plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/async/AsyncPartsHandler.java

```diff
@@ -142,7 +144,9 @@ private static void uploadPart(
             () -> s3AsyncClient.uploadPart(
                 uploadPartRequest,
                 AsyncRequestBody.fromInputStream(
-                    inputStreamContainer.getInputStream(),
+                    // Buffered stream is needed to allow mark and reset ops during IO errors so that only buffered
+                    // data can be retried instead of retrying whole file by the application.
+                    new BufferedInputStream(inputStreamContainer.getInputStream(), (int) (ByteSizeUnit.MB.toBytes(1) + 1)),
```

Comment:
should we use `128KB +1` size ?

Reply:
```
        super(in);
        if (size <= 0) {
            throw new IllegalArgumentException("Buffer size <= 0");
        }
        buf = new byte[size];
```

Looks like we do allocate `size` every time we create a `BufferedInputStream` , so better to allocate the minimum here.

Reply:
It was still producing invalid mark error even with 1MB. So, just kept 1MB to at least allow small file uploads being retried.